src/cryptonote_core/service_node_list.cpp

```diff
@@ -2383,6 +2525,9 @@ namespace service_nodes
     if (derived_x25519_pubkey)
       x25519_to_pub[derived_x25519_pubkey] = {proof.pubkey, now};
 
+    if (x25519_pkey)
+      *x25519_pkey = derived_x25519_pubkey;
```

Comment:
See comment elsewhere: I think we should take a `LokiMQ& lmq` in this function and then call update_active_sns() on it from here *if* the ed25519 key changed from what we had stored.